net2cog/netcdf_convert.py

```diff
@@ -37,152 +35,144 @@ def __init__(self, msg):
         super().__init__(msg)
 
 
-def run_command(command, work_dir):
-    """
-    A simple utility to execute a subprocess command.
-    """
-    try:
-        out_call = check_call(command, stderr=subprocess.STDOUT, cwd=work_dir)
-        return out_call
-    except subprocess.CalledProcessError as err:
-        LOGGER.error("command '%s' return with error (code %s): %s",
-                     err.cmd, err.returncode, err.output)
-        raise
-
+def _rioxr_swapdims(netcdf_xarray):
+    netcdf_xarray.coords['y'] = ('lat', netcdf_xarray.lat)
+    netcdf_xarray.coords['x'] = ('lon', netcdf_xarray.lon)
 
-def check_dir(fname):
-    """
-    To return filename and path without file extension
-    """
-    file_name = fname.split('/')
-    rel_path = pjoin(*file_name[-2:])
-    file_wo_extension, _ = splitext(rel_path)
-    return file_wo_extension
-
-
-def get_gtiff_name(output_file):
-    """
-    To create tmp filename to convert to COG and create a filename
-    just as source but without '.TIF' extension
-    """
-    outf = os.path.basename(output_file)
-    dir_path = dirname(output_file)
-    rel_path = check_dir(outf)
-    out_fname = pjoin(dir_path, rel_path)
-    if not exists(out_fname):
-        os.makedirs(out_fname)
-    return pjoin(out_fname, rel_path)
+    return netcdf_xarray.swap_dims({'lat': 'y', 'lon': 'x'})
 
 
-def _write_cogtiff(out_f_name, nc_xarray):
+def _write_cogtiff(
+    output_directory: str,
+    nc_xarray: xr.Dataset,
+    variable_name: str,
+    input_file_basename: str,
+    logger: Logger,
+) -> str | None:
     """
-    This function converts each variable inside a NetCDF file into a
+    This function converts a variable inside a NetCDF file into a
     cloud optimized geotiff.
 
     Parameters
     ----------
-    out_f_name : string
-        Path to temp gtiff filename excluding file extension
+    out_directory : str
+        Path to temporary directory where output GeoTIFFs will be stored before
+        being staged in S3.
         example :/home/dockeruser/converter/podaac/netcdf_converter/temp/
             netcdf_converter/
             RSS_smap_SSS_L3_8day_running_2020_037_FNL_v04.0_test
-    nc_xarray : xarray dataset
+    nc_xarray : xarray.Dataset
         xarray dataset loaded from NetCDF file
+    variable_name: str
+        Name of the variable as passed in to Harmony.
```

Comment:
nit picky but shouldn't mention Harmony in the non-harmony module. Intent is that this `netcdf_convert` module could be used independently of Harmony.

Reply:
Updated in [279641](https://github.com/podaac/net2cog/pull/38/commits/279641950a6c7edafac149ae76a9749be638432d)